text/0000-better-env-handling.md

```diff
@@ -0,0 +1,89 @@
+- Start Date: 2020-04-23
+- Relevant Team(s): Ember.js, Learning
+- RFC PR: [#619](https://github.com/emberjs/rfcs/pull/619)
+- Tracking: (leave this empty)
+
+# Better environment handling
+
+## Summary
+
+This RFC proposes to add a tiny module similar to `@glimmer/env`, to allow to more easily work with common environment questions.
+
+## Motivation
+
+This RFC tries to solve two common problems:
+
+1 - **Handling test-specific code**.
+
+The old way of doing this, `Ember.testing`, is not really ideal and recommended anymore.
+
+Example use cases where you might want to use this:
+
+- Disabling polling code in tests
+- Preventing test-breaking code like `window.reload()`
+- Preventing log output in tests
+
+2 - **Handling debug-specific (=non-production) code**.
+
+You can use `import { DEBUG } from "@glimmer/env"` for this, or `import { runInDebug } from "@ember/debug"`, although the later [seems to be buggy as of now](https://github.com/emberjs/ember.js/issues/18912).
+
+An exmaple use case where you might want to use this:
+
+- Auto-signing in a mocked user in development environment
+
+This RFC proposes to provide a more more streamlined, approved and documented way to handle these cases, by introducing an `@ember/env` namespace which can be used like this:
+
+```js
+import { DEBUG, TESTING } from "@ember/env";
+```
+
+## Detailed design
+
+```js
+import { DEBUG } from "@ember/env";
+```
+
+This works exactly as `import { DEBUG } from "@glimmer/env"`.
+The reasoning for moving this into the `@ember` namespace is that it is not clear why this should be imported from glimmer, which is the rendering engine.
```

Comment:
`@glimmer/*` is not just about the rendering engine though we haven't _really_ done a great job of explaining how it fits into things. For example `@glimmer/tracking` is not really fundamentally tied to rendering.

Reply:
I am also fine with that! I will update the RFC accordingly, which will then suggest adding a `TESTING` export from `@glimmer/env`, plus this module to the API docs.